contracts/oracles/chainlink/ChainlinkUSDMultiOracle.sol

```diff
@@ -104,3 +105,57 @@ contract ChainlinkUSDMultiOracle is IOracle, AccessControl, Constants {
         updateTime = (updateTime1 < updateTime2) ? updateTime1 : updateTime2;
     }
 }
+
+/**
+ * @title ChainlinkL2USDMultiOracle: ChainlinkUSDMultiOracle that's safe to use on L2
+ * @notice Chainlink recommends checking the sequencer status on some L2 networks to avoid
+  reading stale data
+  
+  https://docs.chain.link/docs/l2-sequencer-flag/
+ */
+contract ChainlinkL2USDMultiOracle is ChainlinkUSDMultiOracle {
+    using CastBytes32Bytes6 for bytes32;
+
+    FlagsInterface public chainlinkFlags;
+    // https://docs.chain.link/docs/l2-sequencer-flag/
+    address internal constant FLAG_ARBITRUM_SEQ_OFFLINE =
+        address(bytes20(bytes32(uint256(keccak256('chainlink.flags.arbitrum-seq-offline')) - 1)));
+
+    constructor(FlagsInterface flags) {
+        require(address(flags) != address(0), 'FlagsInterface has to be set');
+        chainlinkFlags = flags;
+    }
+
+    function assertCanGetPrice() private view {
+        bool isRaised = chainlinkFlags.getFlag(FLAG_ARBITRUM_SEQ_OFFLINE);
+        if (isRaised) {
+            // If flag is raised we shouldn't perform any critical operations
+            revert('Chainlink feeds are not being updated');
+        }
+    }
+
+    /// @dev Convert amountBase base into quote at the latest oracle price.
+    function peek(
+        bytes32 baseId,
+        bytes32 quoteId,
+        uint256 amountBase
+    ) external view virtual override returns (uint256 amountQuote, uint256 updateTime) {
+        assertCanGetPrice();
+        if (baseId == quoteId) return (amountBase, block.timestamp);
+
+        (amountQuote, updateTime) = _peekThroughUSD(baseId.b6(), quoteId.b6(), amountBase);
+    }
+
+    /// @dev Convert amountBase base into quote at the latest oracle price, updating state if necessary. Same as `peek` for this oracle.
+    function get(
+        bytes32 baseId,
+        bytes32 quoteId,
+        uint256 amountBase
+    ) external virtual override returns (uint256 amountQuote, uint256 updateTime) {
+        assertCanGetPrice();
+
+        if (baseId == quoteId) return (amountBase, block.timestamp);
+
+        (amountQuote, updateTime) = _peekThroughUSD(baseId.b6(), quoteId.b6(), amountBase);
+    }
+}
```

Comment:
As before:
```
    function get(bytes32 baseId, bytes32 quoteId, uint256 amountBase)
      external virtual override
      onlyFresh
      returns (uint256 amountQuote, uint256 updateTime)
    {
      return super.get(baseId, quoteId, amountBase);
    }
```

Reply:
nope, can't do that for `external` functions